### ISSUE TITLE
check if backup file exist in E: (Windows)

### ISSUE THREAD

**Issue by annbabich** [newrelic/nri-flex#32]
I'm about to pull all my hair out. I can't figure out how to check if backup file exits in E:\Backup\
Please, help!

**Comment by cah-zack-mutchler** [comment]
This should be marked as a bug.

Currently; the passing of a standard Windows filepath into the YAML config for Flex does not function as expected. 

Example: `D:\TEMP\NRI-Flex\examples`

It seems the problem is the `\` special character. The Go package is adding an escape to the path, but not translating that back to "normal" once it is passed into Windows CMD. (Identified by executing the YAML with the `--verbose` parameter)

Current Work-Around: 

1) Setup PowerShell script to gather data
    -- This script must be saved in a file path without spaces
    -- The "split" attribute doesn't seem to work in this situation, so you need to export the results of your script as a JSON output for Flex to read the results
2) Execute this script in your YAML config


example YAML: 
```YAML
name: windowsFileCount
apis:
  - name: windowsFileCount
    commands:
      # Note: You cannot have spaces in the file path here
      - run: powershell -file D:\TEMP\NewRelicFlexScripts\windowsFileCount.ps1
```

example PowerShell: 
```PowerShell
#region Top of Script

<#
.SYNOPSIS
    Outputs Directory, File Pattern, and Count of Matches

.DESCRIPTION
    Uses the New Relic Flex On-Host Integration
    Built for the "windowsFileCount.yml" Flex Configuration

.NOTES
    Version:        1.0
    Author:         Zack Mutchler
    Creation Date:  05/30/2019
    Purpose/Change: Initial Script development
#>

#endregion

#####-----------------------------------------------------------------------------------------#####

# Set your target directory
$directory = "E:\Backup"

# Set the file name pattern match (this will match on the file extension as well)
# Asterisk ( * ) is the wildcard used here
# Examples:
#   - "*http*" ; find any files with "http" anywhere in the name
#   - "*.log" ; find any files ending in ".log"
#   - "testing.log" ; find any files matching the exact name "testing.log"
$filePattern = "*"


# Count the number of files found matching the supplied pattern (Non-Recursive)
$directory | Get-ChildItem | Where-Object { $_.PsIsContainer -eq $false -and $_.Name -like $filePattern } | Measure-Object | Select-Object -Property @{ expression = { $directory }; name = "directoryName" },@{ expression = { $filePattern }; name = "filePattern" },@{ expression = { $_.Count }; name = "fileCount" } | ConvertTo-Json

# <OPTIONAL> Count the number of files found matching the supplied pattern (Recursive)
#$directory | Get-ChildItem -Recurse | Where-Object { $_.PsIsContainer -eq $false -and $_.Name -like $filePattern } | Measure-Object | Select-Object -Property @{ expression = { $directory }; name = "directoryName" },@{ expression = { $filePattern }; name = "filePattern" },@{ expression = { $_.Count }; name = "fileCount" } | ConvertTo-Json
```

**Comment by annbabich** [comment]
Hi Zac,You made my day. I really appreciate you help .Thank you so muchAnya
    On Wednesday, October 30, 2019, 11:51:13 PM EDT, Zack Mutchler <notifications@github.com> wrote:  
 
 
This should be marked as a bug.

Currently; the passing of a standard Windows filepath into the YAML config for Flex does not function as expected.

Example: D:\TEMP\NRI-Flex\examples

It seems the problem is the \ special character. The Go package is adding an escape to the path, but not translating that back to "normal" once it is passed into Windows CMD. (Identified by executing the YAML with the --verbose parameter)

Current Work-Around:
   
   - Setup PowerShell script to gather data   
-- This script must be saved in a file path without spaces   
-- The "split" attribute doesn't seem to work in this situation, so you need to export the results of your script as a JSON output for Flex to read the results
   - Execute this script in your YAML config

example YAML:
name: windowsFileCount
apis:
  - name: windowsFileCount
    commands:
      # Note: You cannot have spaces in the file path here
      - run: powershell -file D:\TEMP\NewRelicFlexScripts\windowsFileCount.ps1
example PowerShell:
#region Top of Script

<#
.SYNOPSIS
    Outputs Directory, File Pattern, and Count of Matches

.DESCRIPTION
    Uses the New Relic Flex On-Host Integration
    Built for the "windowsFileCount.yml" Flex Configuration

.NOTES
    Version:        1.0
    Author:         Zack Mutchler
    Creation Date:  05/30/2019
    Purpose/Change: Initial Script development
#>

#endregion

#####-----------------------------------------------------------------------------------------#####

# Set your target directory
$directory = "E:\Backup"

# Set the file name pattern match (this will match on the file extension as well)
# Asterisk ( * ) is the wildcard used here
# Examples:
#   - "*http*" ; find any files with "http" anywhere in the name
#   - "*.log" ; find any files ending in ".log"
#   - "testing.log" ; find any files matching the exact name "testing.log"
$filePattern = "*"


# Count the number of files found matching the supplied pattern (Non-Recursive)
$directory | Get-ChildItem | Where-Object { $_.PsIsContainer -eq $false -and $_.Name -like $filePattern } | Measure-Object | Select-Object -Property @{ expression = { $directory }; name = "directoryName" },@{ expression = { $filePattern }; name = "filePattern" },@{ expression = { $_.Count }; name = "fileCount" } | ConvertTo-Json

# <OPTIONAL> Count the number of files found matching the supplied pattern (Recursive)
#$directory | Get-ChildItem -Recurse | Where-Object { $_.PsIsContainer -eq $false -and $_.Name -like $filePattern } | Measure-Object | Select-Object -Property @{ expression = { $directory }; name = "directoryName" },@{ expression = { $filePattern }; name = "filePattern" },@{ expression = { $_.Count }; name = "fileCount" } | ConvertTo-Json

**Comment by Kav91** [comment]
Issue is resolved with the GA 1.0.0 release of Flex, you are able to specify the shell eg. powershell and avoid the escaping issue.
